sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueServiceClient.java

```diff
@@ -342,8 +413,20 @@ public QueueServiceProperties getProperties() {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<QueueServiceProperties> getPropertiesWithResponse(Duration timeout, Context context) {
-        Mono<Response<QueueServiceProperties>> response = client.getPropertiesWithResponse(context);
-        return StorageImplUtils.blockWithOptionalTimeout(response, timeout);
+        Context finalContext = context == null ? Context.NONE : context;
+        try {
+            Supplier<ResponseBase<ServicesGetPropertiesHeaders, QueueServiceProperties>> operation =
+                () -> this.azureQueueStorage.getServices().getPropertiesWithResponse(null, null, finalContext);
+
+            ResponseBase<ServicesGetPropertiesHeaders, QueueServiceProperties> response = timeout != null
+                ? THREAD_POOL.submit(operation::get).get(timeout.toMillis(), TimeUnit.MILLISECONDS) : operation.get();
```

Comment:
Given the wide usage of this pattern we should think about making this a utility method. And possibly think about making that utility method handle try/catching with logging as well.